### PR TITLE
[Feature] New privilege: support dml&load stmt

### LINKED ISSUE #13336: Support dml & load statements in new RBAC privilege framework
## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13336

## Problem Summary(Required) ：

Support dml&load stmt in new privilege framework

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Review by gengjun-git — DISMISSED** [review]

**Comment by nshangyiming** [comment]
plz change pr title with "[Feature]"

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 1]
```diff
@@ -268,7 +268,19 @@ public enum ErrorCode {
     ERR_WRONG_LABEL_NAME(5082, new byte[] {'4', '2', '0', '0', '0'},
             "Incorrect label name '%s'"),
     ERR_CHANGE_TO_SSL_CONNECTION_FAILED(5083, new byte[] {'4', '2', '0', '0', '0'},
-            "Change to ssl connection failed");
+            "Change to ssl connection failed"),
+
+    ERR_PRIVILEGE_ROUTINELODE_JOB_NOT_FOUND(5084, new byte[] {'4', '2', '0', '0', '0'},
+            "Routine load job not found when check privilege"),
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:274]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 2]
```diff
@@ -162,6 +219,50 @@ static void checkStmtNodePrivilege(ConnectContext context) {
         }
     }
 
+    static void checkOperateLoadPrivilege(ConnectContext context, String dbName, String label) {
+        GlobalStateMgr globalStateMgr = context.getGlobalStateMgr();
+        Database db = globalStateMgr.getDb(dbName);
+        if (db == null) {
+            ErrorReport.reportSemanticException(ErrorCode.ERR_PRIVILEGE_DB_NOT_FOUND);
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:226]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 3]
```diff
@@ -162,6 +219,50 @@ static void checkStmtNodePrivilege(ConnectContext context) {
...
+        }
+        List<LoadJob> loadJobs = globalStateMgr.getLoadManager().
+                                                getLoadJobsByDb(db.getId(), label, false);
+        List<String> forbiddenInsertTableList = new ArrayList<>();
+        List<String> forbiddenUseResourceList = new ArrayList<>();
+        loadJobs.forEach(loadJob -> {
+            try {
+                if (!PrivilegeManager.checkResourceAction(context, loadJob.getResourceName(),
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:234]: done

[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:234]: we should check table's insert and resource's usage

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 4]
```diff
@@ -182,6 +283,106 @@ public Void visitInsertStatement(InsertStmt statement, ConnectContext session) {
...
+                                                    context.getQualifiedUser(),
+                                                    context.getRemoteIP(),
+                                                    forbiddenInsertTableList.toString());
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitShowLoadStatement(ShowLoadStmt statement, ConnectContext context) {
+            // No authentication required
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:376]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 5]
```diff
@@ -106,6 +125,44 @@ public static void checkTableAction(ConnectContext context,
...
 
+    static void checkRoutineLoadActionByLabel(ConnectContext context,
+                                              String dbName, String labelName,
+                                              PrivilegeType.TableAction action) {
+        RoutineLoadJob job = null;
+        String tableName = null;
+        try {
+            job = context.getGlobalStateMgr().getRoutineLoadManager().getJob(dbName, labelName);
+        } catch (MetaNotFoundException e) {
+            ErrorReport.reportSemanticException(ErrorCode.ERR_PRIVILEGE_ROUTINELODE_JOB_NOT_FOUND);
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:136]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 6]
```diff
@@ -106,6 +125,44 @@ public static void checkTableAction(ConnectContext context,
...
+        }
+        try {
+            tableName = job.getTableName();
+        } catch (MetaNotFoundException e) {
+            ErrorReport.reportSemanticException(ErrorCode.ERR_PRIVILEGE_TABLE_NOT_FOUND);
+        }
+        checkTableAction(context, dbName, tableName, action);
+    }
+
+    static void checkTableAction(ConnectContext context,
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:149]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 7]
```diff
@@ -106,6 +125,44 @@ public static void checkTableAction(ConnectContext context,
...
+                                              String dbName, String tableName,
+                                              PrivilegeType.TableAction action) {
+        if (null == action) {
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:152]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 8]
```diff
@@ -182,6 +283,106 @@ public Void visitInsertStatement(InsertStmt statement, ConnectContext session) {
             return null;
         }
 
+        @Override
+        public Void visitUpdateStatement(UpdateStmt statement, ConnectContext context) {
+            // No authentication required
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:288]: done

**Review by nshangyiming — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 113 / 168 (67.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/LoadStmtAnalyzer.java|1|7|14.29%|[100, 101, 103, 104, 105, 106]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadManager.java|3|15|20.00%|[469, 484, 486, 488, 489, 492, 493, 495, 496, 497, 504, 517]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|19|26.32%|[1019, 1020, 1021, 1023, 1024, 1025, 1026, 1028, 1030, 1031, 1089, 1093, 1094, 1095]|
|:large_blue_circle:|com/starrocks/sql/ast/DataDescription.java|4|10|40.00%|[594, 595, 596, 604, 605, 606]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|84|101|83.17%|[156, 157, 159, 160, 163, 164, 165, 167, 175, 176, 179, 183, 184, 338, 357, 358, 465]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadManager.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|5|5|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Comment by waittttting** [comment]
run starrocks_admit_test

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Pull request merged**